pilot/pkg/model/jwks_resolver.go

```diff
@@ -247,19 +251,22 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 		log.Errorf("Failed to jwks URI from %q: %v", issuer, err)
 	} else {
 		var resp []byte
-		resp, err = r.getRemoteContentWithRetry(jwksURI, networkFetchRetryCountOnMainFlow)
+		resp, err = r.getRemoteContentWithRetry(jwksURI, 0)
 		if err != nil {
 			log.Errorf("Failed to fetch public key from %q: %v", jwksURI, err)
+			// running the process of fetching public key in background
+			r.refreshTicker.Stop()
+			r.refreshIntervalOnFailure = time.Second
+			r.refreshInterval = time.Second
+			go r.refresher()
```

Comment:
I think you should reuse the same refresher routine instead of starting a new one.

You can do some refactor to let you just add the new jwksUri to the refresher. The fetching/retry/etc. (e.g. internal) should be internal logic to the refresher and not exposed here.

Reply:
will that be okay if I add some extra variables in `JwksResolver` struct ?https://github.com/istio/istio/blob/58bf51baa71909c1b16a90d7d5acaf5b2c499a9a/pilot/pkg/model/jwks_resolver.go#L116

Reply:
And I have to do something with `refreshTicker` in this method as `refresher()` method is always running. Will try to rest the ticker like `r.refreshTicker.Reset(time.Second)`.